packages/nox/test/tpetra/Brusselator/Example.cpp

```diff
@@ -0,0 +1,84 @@
+
+// AM: REDO THE HEADER!!!
+
+#include <Tpetra_Core.hpp>
+
+#include <Teuchos_StandardCatchMacros.hpp>
+
+// User's application specific files
+#include "Problem_Interface.hpp"
+#include "Brusselator.hpp"
+
+template <typename ScalarType>
+int run(int argc, char *argv[])
+{
+  using Teuchos::RCP;
+  using Teuchos::rcp;
+  using Teuchos::ParameterList;
+
+  using ST = typename Tpetra::Vector<ScalarType>::scalar_type;
+  using LO = typename Tpetra::Vector<>::local_ordinal_type;
+  using GO = typename Tpetra::Vector<>::global_ordinal_type;
+  
+  using tvector_t = typename Tpetra::Vector<ST, LO, GO>;
+
+  Teuchos::GlobalMPISession mpiSession(&argc, &argv, &std::cout);
+  auto Comm = Tpetra::getDefaultComm();
+  const int MyPID = Comm->getRank();
+  const int NumProc = Comm->getSize();
+
+  bool verbose = true;
+  bool success = false;
+
+  try {
+    // AM: Before Epetra timer, redo?
+
+    // Check for verbose option
+    if (argc > 1)
+      if (argv[1][0]=='-' && argv[1][1]=='v')
+        verbose = true;
+    
+    // Set the problem size (1000 elements)
+    int NumGlobalNodes = 1001;
+
+    // The number of unknowns must be at least equal to the
+    // number of processors.
+    if (NumGlobalNodes < NumProc) {
+      std::cout << "numGlobalNodes = " << NumGlobalNodes
+                << " cannot be < number of processors = " << NumProc << std::endl;
+      std::cout << "Test failed!" << std::endl;
+      exit(1);
+    }
+
+    // Create the Brusselator problem class. This creates all required
+    // Tpetra objects for the problem and allows calls to the
+    // function (F) and Jacobian evaluation routines.
+    Brusselator<ScalarType>::OverlapType OType = Brusselator<ScalarType>::NODES;                            // AM: TOFIX
```

Comment:
Issue:
```
Brusselator/Example.cpp:56:5: error: need ‘typename’ before ‘Brusselator<ScalarType>::OverlapType’ because ‘Brusselator<ScalarType>’ is a dependent scope
   56 |     Brusselator<ScalarType>::OverlapType OType = Brusselator<ScalarType>::NODES;
```

Reply:
If you replace `ScalarType` with `ST`, does it work?